core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/Changeset.java

```diff
@@ -127,7 +127,25 @@ public void close() throws SailException {
 		refbacks = null;
 		prepend = null;
 		observed = null;
-		approved = null;
+		try {
+			if (approved instanceof AutoCloseable) {
+				try {
+					((AutoCloseable) approved).close();
+				} catch (Exception e) {
+					throw new SailException(e);
+				}
+			}
+			approved = null;
+		} finally {
+			if (deprecated instanceof AutoCloseable) {
+				try {
+					((AutoCloseable) deprecated).close();
+				} catch (Exception e) {
+					throw new SailException(e);
+				}
+			}
+			deprecated = null;
+		}
 		deprecated = null;
 		approvedContexts = null;
 		deprecatedContexts = null;
```

Comment:
Could you wrap your try-finally in another try-finally and in the finally set all the fields to null? That way we are sure to always clear up and references that could slow down GC.

Reply:
I've changed this.